content/en/posts/mesa-optimizers.md

```diff
@@ -0,0 +1,40 @@
+(semi-)supervised way, arguably they often just end up following some behaviours that "happen" to score highly on the
+training target. This doesn't mean they have any kind of explicit internal representation of that training target, let
+alone they deliberately try to optimize for it. On the other hand, there clearly is some optimization involved in the training of AIs, often in the form of stochastic gradient descent (SGD). So usually there indeed is an optimizer at play, but the optimization tends to happen *during training* rather than *during inference* (= when the model is deployed in the real world).
+
+Does this mean Bostrom and Yudkowsky were misguided in their worries, and trained AIs will be harmless, since they themselves are not optimizers? Unfortulately not, as this is where mesa optimizers enter the playing field. "Mesa" is simply the opposite of "meta", so something like "one level down". The idea is that a training process (in this context usually called the *base optimizer*) could potentially select for programs that themselves are optimizers (the *mesa optimizer*). Theoretically this makes sense: an agent that is actively optimizing for some training target has a good chance of being among the best agents when it comes to actually achieving said training target. Whereas non-optimizer agents likely perform worse in comparison.
+
+However, the training loop is not some omniscient or omnipotent entity that just happens to find the best possible program or agent. It is still a relatively simple process, that usually starts with some randomized initialization of model weights, and then performs a greedy search process (such as SGD), iteratively improving the model weights. It's unclear under which circumstances exactly such a process would actually *find* a set of model weights that would yield a mesa optimizer for any given initialization.
+
+Yet, despite these uncertainties, there are at least two reasons to take this idea quite seriously:
+
+1. Even if currently training loops mostly don't end up creating mesa optimizers and maybe even lack the capabilities to identify such programs in the vast multi-dimensional search space of model weights, we just need to look one level up: behind each ML training process, there's a bunch of **human researchers** who constantly tweak the training regime in order to come up with the most capable possible models. These humans **have an incentive to keep adjusting the training process until it will eventually be able to create mesa optimizers**.
+2. **Humans could be considered exemplary for mesa optimization**: a simple outside process, namely evolution, has run an optimization on living beings, selecting inclusive genetic fitness, and eventually (quite misaligned, from the point of view of evolution) optimizers emerged in the form of humans.
```

Comment:
Making it a little simpler this could be used as a "basic" example e.g. the evolutionary drive for high calories vs. our drive to find the closest fast food place.

Reply:
I now added contraception as an example of humans being "misaligned mesa optimizers" to the basic part:

> One example of mesa optimization is how evolution eventually led to humans, who are (in certain ways) optimizers themselves: we tend to deliberately shape the world so as to move it closer to our preferences.
> This example also shows why the possibility mesa optimizers is concerning: humans are in many ways not aligned with the "goals" of evolution. Contraception is a prime example of this. If machine learning eventually leads to similarly strong mesa optimizers, coupled with similarly misaligned goals, this can lead to very undesirable outcomes.